OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -1364,32 +1364,63 @@ extension OmnipodPumpManager: PumpManager {
                 state.bolusEngageState = .engaging
             })
 
-            var podStatus: StatusResponse
-
-            do {
-                podStatus = try session.getStatus()
-            } catch let error {
-                completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
-                return
-            }
-
             // If pod suspended, resume basal before bolusing
-            if podStatus.deliveryStatus == .suspended {
+            if case .some(.suspended) = self.state.podState?.suspendState {
                 do {
                     let scheduleOffset = self.state.timeZone.scheduleOffset(forDate: Date())
                     let beep = self.confirmationBeeps
-                    podStatus = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
+                    _ = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
                 } catch let error {
+                    self.log.error("enactBolus: error resuming suspended pod: %s", String(describing: error))
                     completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
                     return
                 }
             }
 
-            guard !podStatus.deliveryStatus.bolusing else {
-                completion(.failure(PumpManagerError.communication(PodCommsError.unfinalizedBolus)))
+            var attemptUncertainBolusReconciliation = false
+            var finalizeFinishedDoses = false
+            let preError = self.setStateWithResult({ (state) -> PodCommsError? in
+                if case .some(.suspended) = state.podState?.suspendState {
+                    self.log.error("enactBolus: failing with suspended pod")
+                    return .podSuspended
+                }
+                if let unfinalizedBolus = state.podState?.unfinalizedBolus {
+                    if unfinalizedBolus.scheduledCertainty == .uncertain {
+                        self.log.info("enactBolus: doing getStatus with uncertain bolus scheduled certainty")
+                        attemptUncertainBolusReconciliation = true
+                        return nil
+                    }
+                    if !unfinalizedBolus.isFinished {
```

Comment:
This check does not guarantee the pod is actually done bolusing. It's an estimate. If our estimated time to complete the bolus is too short, and the pod is still bolusing, then we could trigger a fault, no?

Reply:
This is correct but has never happened in all its previous use. Reworked logic to use new isBolusPositivelyFinished variable that adds 20% more time as pod will fault if any pulse takes more than 20% longer than nominal to complete. If the isFinished is true but isBolusPositivelyFinished is false, then a getStatus call will be used to verify if the bolus is indeed completed to avoid any possible pod fault.